src/main/java/hudson/tasks/test/TestResultProjectAction.java

```diff
@@ -118,18 +119,28 @@ protected LinesChartModel createChartModel() {
             TestResultImpl pluggableStorage = storage.load(lastCompletedBuild.getParent().getFullName(), lastCompletedBuild.getNumber());
             return new TestResultTrendChart().create(pluggableStorage.getTrendTestResultSummary());
         }
-        
-        return new TestResultTrendChart().create(createBuildHistory(lastCompletedBuild), new ChartModelConfiguration());
+
+        TestResultActionIterable buildHistory = createBuildHistory(lastCompletedBuild);
+        if (buildHistory == null) {
+            return new LinesChartModel();
```

Comment:
Might be better to have a new implementation with "No Build History" text later.